frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -701,40 +702,59 @@ export default ({
 
       // NOTE: save messages in the browser storage, but not more than CHATROOM_MAX_ARCHIVE_ACTION_PAGES pages of events
       if (this.latestEvents.length >= CHATROOM_MAX_ARCHIVE_ACTION_PAGES * unit) {
-        sbp('gi.db/archive/delete', this.getArchiveKeyFromChatRoomId(chatRoomId))
+        sbp('gi.db/archive/delete', this.archiveKeyFromChatRoomId(chatRoomId))
       } else if (to !== this.messageState.prevTo || from !== this.messageState.prevFrom) {
         // this.currentChatRoomId could be wrong when the channels are switched very fast
         // so it's good to initiate using input parameter chatRoomId
-        sbp('gi.db/archive/save', this.getArchiveKeyFromChatRoomId(chatRoomId), JSON.stringify(this.latestEvents))
+        sbp('gi.db/archive/save', this.archiveKeyFromChatRoomId(chatRoomId), JSON.stringify(this.latestEvents))
       }
     },
-    getArchiveKeyFromChatRoomId (chatRoomId) {
+    archiveKeyFromChatRoomId (chatRoomId) {
       const curChatRoomId = chatRoomId || this.currentChatRoomId
-      return `messages/${curChatRoomId}`
+      return `messages/${this.ourUsername}/${curChatRoomId}`
```

Comment:
Just to be clear: in the future changes like this will require a migration step when the app is starting up. Otherwise without it you'll be polluting their local IndexedDB with data that is then "stuck" there

Reply:
Correct. We should take care of the `bad-formatted` data in the storage either.

And do you think it's better way if we do this way?
- Do not mention username inside the key every time we archive
- After login, we can update these sbp functions (`'gi.db/archive/delete'`, `'gi.db/archive/save'`, `'gi.db/archive/load'`) and set key prefix of `loggedIn.username`?

Reply:
I mean we can redefine those three functions after the user logs in so that we don't need to take care of username in making keys.

Reply:
That's an interesting idea. If we only ever use those functions when logged in, then yes we could do that. I'm not sure if we do.